server/components/tags/tag-model.js

```diff
@@ -0,0 +1,15 @@
+const mongoose = require("mongoose");
+
+const tagSchema = new mongoose.Schema({
+	name: {
+		type: String,
+		required: true
+	},
+	children: {
+		type: [String]
+	}
+});
+
+tagSchema.index({name: "text"});
```

Comment:
Does this mean that we have an index which allows us to query tags based on their name?
Also, I think we should add indices to the Post model (probably for the `title`, `authorId` and `replyId` properties).

Reply:
Yes^^ 
Actually you can already query whatever model with whatever field you like, the index creation allows to make this happens faster ( it creates additional 'tables ' sorted using the specified index so the lookup is much faster if i remember correctly ). The downside is that the size of the db grows ( not a big issue in this case i guess ).
About the other indexes no problem! 
Even if...you just let me think, the title of a post is required, but which title has a reply? Shoudn't it be optional and related to the isQquestion field?

Reply:
> the title of a post is required, but which title has a reply? Shoudn't it be optional and related to the isQuestion field?

Yeah, you're right, I didn't think about this case ahead of time.
It would be great if you could remove the `required` field from the `title` property of the Post schema. Also, in the `createPost` method, we need to verify that if the given post is also a question, it should have a non-empty title.

Reply:
Ok. i'ìll update it this afternoon^^

Reply:
So,
after a couple of hours trying to update this pr i opened a new one o.o 
I can't understand why it doesn't allow me to update this, on pull says  ' already up to date' and on push keeps saying ' a tip of the branch is ahead [....]' .
That pr _should_ be at the same level of this one, check it out please :/

Reply:
Since I pushed commits to the `feature/post-component` branch, you need to pull this branch from your fork before you can push to it.